src/annotator/guest.js

```diff
@@ -439,10 +432,16 @@ export default class Guest {
       annotations => annotations.forEach(annotation => this.anchor(annotation))
     );
 
-    // Discover and connect to the sidebar frame. All RPC events must be
-    // registered before creating the channel.
-    const sidebarPort = await this._portFinder.discover('sidebar');
-    this._sidebarRPC.connect(sidebarPort);
+    // Connect to sidebar and send document info/URIs to it.
+    //
+    // RPC calls are deferred until a connection is made, so these steps can
+    // complete in either order.
+    this._portFinder.discover('sidebar').then(port => {
+      this._sidebarRPC.connect(port);
+    });
+    this.getDocumentInfo().then(metadata =>
+      this._sidebarRPC.call('documentInfoChanged', metadata)
+    );
```

Comment:
Why not continue sending the `error` and `data` (following the Node.js convention)?

```suggestion
    this.getDocumentInfo().then(metadata =>
      this._sidebarRPC.call('documentInfoChanged', null, metadata)
    ).catch(error => 
      this._sidebarRPC.call('documentInfoChanged', 'Failed to get the document information', null)
    );
```

Alternatively, `metadata` could be a `string` (for the text of the error) or the current data structure. If the text is meaningless, then a `null` could indicate the error.

In the future, once `Exceptions` are able to be transmitted among frames, it could be an `Exception` or the current data structure.